src/pyswi/swi_ts/swi_ts.py

```diff
@@ -137,8 +138,31 @@ def swi_error_prop(ssm, t_value, t_noise, swi_error, gain_in=None, nan=-9999.):
                 swi_noise[i, c] = np.nan
                 qflag[i, c] = qflag[i-1][c] * np.exp(-(1. / float(t_value[c])))
 
-    gain_out = {'last_jd': last_jd, 'last_gain': gain_curr, 'last_contr1': contr1_curr,
-                'last_G': G_curr, 'last_JT': JT_curr, 'last_swi': swi[-1], 'last_noise': swi_noise[-1], 'last_qflag': qflag[-1]}
+    # If there are no input ssm values, gain_out should not be updated and should be the same as gain_in.
+    if gain_in and len(ssm[~np.isnan(ssm['sm'])]) == 0:
+        gain_out = gain_in
+        gain_out['last_qflag'] = qflag[-1] # Update to last qflag
+
+    # If there is only 1 input SSM value:
+    elif gain_in and len(ssm[~np.isnan(ssm['sm'])]) == 1:
+        gain_out = {'last_jd': last_jd, 
+                    'last_gain': gain_curr, 
+                    'last_contr1': contr1_curr,
+                    'last_G': G_curr, 
+                    'last_JT': JT_curr,
+                    'last_swi': swi[first_ssm_idx], 
+                    'last_noise': swi_noise[first_ssm_idx], 
+                    'last_qflag': qflag[-1]} # NOTE this is the last qflag.
+    else:
+        gain_out = {'last_jd': last_jd, # Last date where a swi calculation was done
+                    'last_gain': gain_curr, 
+                    'last_contr1': contr1_curr,
+                    'last_G': G_curr, 
+                    'last_JT': JT_curr, 
+                    'last_swi': swi[last_valid_index], 
```

Comment:
should `last_valid_index` have an initial value, eg 0? e.g in case input ssm is empty